src/DotLiquid.Tests/StandardFilterTests.cs

```diff
@@ -2014,10 +2014,11 @@ public void TestWhere_NonStringCompare()
 {% endfor %}",
                 localVariables: Hash.FromAnonymousObject(new { products }));
 
-            // The products array has a price of 7.0f, compare to the string '7.0'
+            // The products array has a price of 7.0f, compare to the string '7'
+            // Expect string != number
             Helper.AssertTemplateResult(
-                expected: "\r\n\r\nCheapest products:\r\n\r\n- Spatula\r\n",
-                template: @"{% assign cheap_products = products | where: ""price"", '7.0' %}
+                expected: "\r\n\r\nCheapest products:\r\n",
+                template: @"{% assign cheap_products = products | where: ""price"", '7' %}

```

Comment:
you are changing the test from explicit float to probably a defaulting to integer. It's ok to have a test with integer comparison to float, but you should have as an additional test then.

Reply:
It's a string, not an integer. And in the new code strings do not equate/match numbers (as per shopify). I changed it from `'7.0'` to match the test just above it (which uses `7`), just to show that it will match the integer but won't match the identical string. We already have a float and an int test. 

https://replit.com/@rrjames/Liquid-Tests shows strings do no equate to numbers.